packages/core/src/combinator/combine.ts

```diff
@@ -0,0 +1,110 @@
+/** @license MIT License (c) copyright 2010 original author or authors */
+
+import { map } from './transform'
+import { empty, containsCanonicalEmpty } from '../source/empty'
+import Pipe from '../sink/Pipe'
+import IndexSink, { IndexedValue } from '../sink/IndexSink'
+import { disposeAll, tryDispose } from '@most/disposable'
+import invoke from '../invoke'
+import { Stream, Sink, Scheduler, Disposable, Time } from '@most/types'
+import { ToStreamsArray } from './variadic'
+
+/**
+ * Combine latest events from two streams
+ * @param f function to combine most recent events
+ * @param stream1
+ * @param stream2
+ * @returns stream containing the result of applying f to the most recent
+ *  event of each input stream, whenever a new event arrives on any stream.
+ */
+export const combine = <A, B, C>(f: (a: A, b: B) => C, stream1: Stream<A>, stream2: Stream<B>): Stream<C> =>
+  combineArray(f, [stream1, stream2])
+
+/**
+* Combine latest events from all input streams
+* @param f function to combine most recent events
+* @param streams most recent events
+* @returns stream containing the result of applying f to the most recent
+*  event of each input stream, whenever a new event arrives on any stream.
+*/
+export const combineArray = <Args extends any[], R>(f: (...args: Args) => R, streams: ToStreamsArray<Args>): Stream<R> =>
+  streams.length === 0 || containsCanonicalEmpty(streams) ? empty()
+    : streams.length === 1 ? map(f as any, streams[0])
+      : new Combine(f, streams)
+
+class Combine<Args extends any[], B> {
```

Comment:
I think these streams should implement the `Stream` interface. Like:

```typescript
class Combine<Args extends any[], B> implements Stream<B> {
  // ...
}
```

It ensures that the class implements the `run` method correctly.

Reply:
done